src/components/SearchInput.tsx

```diff
@@ -1,58 +1,64 @@
 import { SearchIcon } from "./Icons";
 import { useState, useCallback } from "react";
-import { useSearchParams, useNavigate } from "react-router-dom";
+import { useSearchParams, useNavigate, useLocation } from "react-router-dom";
 
 const SearchInput = () => {
   const navigate = useNavigate();
+  const location = useLocation();
   const [searchParams, setSearchParams] = useSearchParams();
   const [searchValue, setSearchValue] = useState(searchParams.get("q") || "");
 
-  const debouncedSearch = useCallback(
-    debounce((query: string) => {
-      if (query) {
-        setSearchParams({ q: query });
-        navigate(`/search?q=${encodeURIComponent(query.trim().toLowerCase())}`);
-      } else {
+  const navigateToSearch = useCallback(
+    (query: string, isCompletedSearch = false) => {
+      const trimmedQuery = query.trim().toLowerCase();
+
+      if (!trimmedQuery) {
+        // Remove search params and navigate to home if query is empty
         setSearchParams({});
-        navigate("/");
+        navigate("/", { replace: true });
+        return;
       }
-    }, 200),
-    [setSearchParams, navigate]
-  );
 
-  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const value = e.target.value;
-    setSearchValue(value);
-    debouncedSearch(value);
-  };
+      // Set the search params with the query
+      // Use replace: true for keypresses (when isCompletedSearch is false)
+      setSearchParams({ q: trimmedQuery }, { replace: !isCompletedSearch });
+
+      // Only navigate if we're not already on the search page
+      if (location.pathname !== "/search") {
+        navigate("/search", {
+          replace: isCompletedSearch || location.pathname === "/search",
```

Comment:
This one should always be false

Reply:
I'll look at it next year, I've been going crazy for the past four hours trying to get this all to behave.